app/components/home/home.js

```diff
@@ -0,0 +1,54 @@
+'use client';
+import { useState } from 'react';
+import InfiniteScroll from 'react-infinite-scroller';
+import Image from 'next/image';
+import { fetchPokemonData } from '../../../api/fetch-pokemon';
+import styles from "./home.module.css";
+
+export const HomePage = () => {
+  const limit = 10;
+  const [pokemon, setPokemon] = useState([]);
+  const [hasMore, setHasMore] = useState(true);
+  const [offset, setOffset] = useState(0);
+  const map = {};
+
+  const loadMore = async () => {
+
+    if(offset in map)
+      return
+
+    map[offset] = true;
+
```

Comment:
Set initialLoad to true
Get the first page's results' from the server side and pass it to HomePage via props.
use the isLoading true/false in the function